src/components/base/FiscalYearStartSelectionBase.ts

```diff
@@ -0,0 +1,111 @@
+import { computed } from 'vue';
+
+import { FiscalYearStart } from '@/core/fiscalyear.ts';
+
+import { useI18n } from '@/locales/helpers.ts';
+
+import { formatMonthDay } from '@/lib/datetime.ts';
+
+import { useUserStore } from '@/stores/user.ts';
+
+export interface FiscalYearStartSelectionBaseProps {
+    modelValue?: number;
+}
+
+export interface FiscalYearStartSelectionBaseEmits {
+    (e: 'update:modelValue', value: number): void;
+}
+
+export function useFiscalYearStartSelectionBase(props: FiscalYearStartSelectionBaseProps, emit?: FiscalYearStartSelectionBaseEmits) {
+    const { getCurrentFiscalYearStart, getLocalizedLongMonthDayFormat } = useI18n();
+    const userStore = useUserStore();
+
+    const getDefaultValue = (): number => {
+        return userStore.currentUserFiscalYearStart || FiscalYearStart.DefaultNumber;
+    };
+
+    const effectiveModelValue = computed<number>(() => {
+        return props.modelValue !== undefined ? props.modelValue : getDefaultValue();
```

Comment:
if `getDefaultValue` is not used anywhere else, is it better to use `getCurrentFiscalYearStart().value` to replace `getDefaultValue`?